float64 x
float64 y
float64 heading
float64 left_width
float64 right_width
float64 speed